operators/src/source/ogr_source.rs

```diff
@@ -396,19 +261,19 @@ pub struct OgrSourceProcessor<G>
 where
     G: Geometry + ArrowTyped,
 {
-    dataset_information: Arc<OgrSourceDataset>,
-    default_geometry: Option<G>,
+    dataset_information: Box<dyn LoadingInfo<OgrSourceDataset, VectorResultDescriptor>>,
     _collection_type: PhantomData<FeatureCollection<G>>,
 }
 
 impl<G> OgrSourceProcessor<G>
 where
     G: Geometry + ArrowTyped,
 {
-    pub fn new(dataset_information: Arc<OgrSourceDataset>, default_geometry: Option<G>) -> Self {
+    pub fn new(
+        dataset_information: Box<dyn LoadingInfo<OgrSourceDataset, VectorResultDescriptor>>,
```

Comment:
> jdroenner commented on 20.01.2021
maybe this could also be a generic. then we just use it with Box if we implement the trait for Box.

Reply:
but then we don't even avoid the indirections but hide them?!

Reply:
but it looks nicer

Reply:
this would go against rust's phliosophy. They added the `dyn` keyword exactly for the reason to make it apparent when there is cost to something.